### PR TITLE
refactor: types for http tracing plugin

### PR DESCRIPTION
**Pull request by kjin**

### CODE REVIEW

File: src/plugins/plugin-http.ts [hunk 1]
```diff
@@ -14,41 +14,65 @@
...
-'use strict';
-var shimmer = require('shimmer');
-var url = require('url');
-var isString = require('is').string;
-var httpAgent = require('_http_agent');
-var semver = require('semver');
-
-function getSpanName(options) {
+import * as httpModule from 'http';
+import {Agent, ClientRequest, ClientRequestArgs, get, request} from 'http';
```
Comment (jinwoo) [inline src/plugins/plugin-http.ts]: You shouldn't import `get` and `request`. That'll make the `http` module loaded at run time before the it's instrumented. You should import only types using this syntax.

Comment (jinwoo) [reply src/plugins/plugin-http.ts]: In other words, you should look at the transpiled .js file and see if the `http` module is `require`'d there. It is a problem if it is.

Comment (kjin) [reply src/plugins/plugin-http.ts]: Good catch -- these imports get erased at compile time because they are only used for types.

Comment (jinwoo) [reply src/plugins/plugin-http.ts]: Ah, I missed that they are being used only for types. Sorry for the noise.

**Comment by kjin** [comment]
Ignoring codecov check for now... it will change drastically in #651 anyways

**Pull request merged**